### ISSUE TITLE
Get OneLink attribution parameters while app is in background 

### ISSUE THREAD

**Issue by Florin-Ion** [AppsFlyerSDK/appsflyer-flutter-plugin#57]
Hi,
We implemented OneLink and we are trying to redirect the user to different pages in the app based on the attribution parameters added to the OneLink custom links. This is working great when the app is closed or when it's installed from said link, as we are getting both the onInstallConversionDataLoaded and onAppOpenAttribution events, but we are not receiving anything if the app is in background and brought in foreground using OneLink. Has anyone encountered this issue before? Is there any way we could achieve this?

This is the flutter doctor -v output:

> [√] Flutter (Channel stable, 1.22.3, on Microsoft Windows [Version 10.0.18363.1198], locale en-US)
    • Flutter version 1.22.3 at D:\Flutter\flutter
    • Framework revision 8874f21e79 (4 weeks ago), 2020-10-29 14:14:35 -0700
    • Engine revision a1440ca392
    • Dart version 2.10.3

> [√] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at C:\Users\Florin\AppData\Local\Android\sdk
    • Platform android-30, build-tools 29.0.3
    • Java binary at: D:\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)
    • All Android licenses accepted.

> [√] Android Studio (version 3.6)
    • Android Studio at D:\Android Studio
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.8052
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)

**Comment by fabeider** [comment]
I'm trying to do the same, but the flutter plugin does not seem to support OneLink.
How did you receive those events?

**Comment by ShaharAF** [comment]
Hi @Florin-Ion, thank you for opening this issue. We are investigating this issue and will fix it in the next plugin version

**Comment by fabeider** [comment]
Hey @ShaharAF thanks for the update.
Do you have a roadmap or date for the next plugin version release?

**Comment by arno-dev** [comment]
Hi, 
I'm currently having a similar issue, below my flutter doctor :
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel unknown, v1.17.0, on Mac OS X 10.15.2 19C57, locale en-US)
 
[!] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses
[!] Xcode - develop for iOS and macOS (Xcode 11.4.1)
    ! CocoaPods 1.7.5 out of date (1.8.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade:
        sudo gem install cocoapods
[✓] Chrome - develop for the web
[!] Android Studio (version 3.6)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
[✓] VS Code (version 1.51.1)
[✓] Connected device (3 available)

```
the method onInstallConversionDataLoaded is being called properly but never onAppOpenAttribution method. I'm setting both values for af_dp and URL scheme fallback on my OneLink.

**Comment by ShaharAF** [comment]
Hi, I'm working on a solution for that issue of the background switch. 
Will release the new version by the end of the week

**Comment by wangxingxing123654** [comment]
> Hi, I'm working on a solution for that issue of the background switch.
> Will release the new version by the end of the week

the  deeplink  cannot get params in flutter ios, can you merge the fixed  ios-link into the master??

**Comment by fabeider** [comment]
Hey guys, any update on this issue?
Updated to v6.0.5+3 and still happening

**Comment by potato16** [comment]
I have the same issue. It works great when the app is closed.

**Comment by w3ggy** [comment]
I have the same issue

**Comment by fabeider** [comment]
Hello, 

Updated to version 6.2.0 and still have the same issue.
Anyone had better luck?

**Comment by GM-appsflyer** [comment]
Hey!
Please try updating to the latest version. The issue should be fixed!

**Comment by misterfourtytwo** [comment]
hey, could you help me please.
i don't know where do i do wrong, i added all three callbacks, enabled them in the initSdk and installed deep_link_value in one link custom links options, but do not get my values from link.
updated right now from 6.2.0 to 6.2.1 but this changed nothing.

**Comment by GM-appsflyer** [comment]
Hey! @misterfourtytwo 
When you're trying to deeplink into the app:
1. Is the app installed/not installed 
2. Is the app being opened?
3. If all of the callbacks are set to true, you should handle all deeplink scenarios on 'onDeepLinking' the unified deeplink callback.
4. Is it happening on iOS/Android?
5. Could you provide native logs when deeplink into the app?

Thanks!

**Comment by misterfourtytwo** [comment]
when it is installed, app always opens, for each url scheme(applink, universal link, uri scheme). i didn't check the case of deferred deeplink cause there is production version in stores, and it is without url schemes.
report i give based on debug version launched/installed from connected pc.
3. so, to actually handle all use cases it is sufficient to register this one callback?
4. callbacks do not trigger both on android and ios.
5. how do i get it? is there a guide how do i extract this from a running dev app? is it enough to just get what appsflyer logs into  console?
this is what i get on deeplink click when app is in bg mode.
![image](https://user-images.githubusercontent.com/17276509/108765893-74673680-7565-11eb-93be-66f4eeeabeeb.png)

**Comment by GM-appsflyer** [comment]
Hey @misterfourtytwo,

Could you provide us with full logs (try using adb logcat | grep "AppsFlyer_"). We would like to see if the native callbacks are being called.
Also, did you make sure to add:
@Override
    public void onNewIntent(Intent intent) {
         super.onNewIntent(intent);
         setIntent(intent);
    }
If the launch mode is [non-standard launch mode](https://developer.android.com/guide/topics/manifest/activity-element#lmode) (which is single by default in flutter).

**Comment by Florin-Ion** [comment]
Hello,

Thank you for your support. We implemented the new 'onDeepLinking' method and we are now getting OneLink data when the app is in background,  but we are not getting the URL that was used in the response like we used to when using "appOpenAttributionStream" or "conversionDataStream". Is there any way we can get that parameter?

Thank you

**Comment by misterfourtytwo** [comment]
> Hey @misterfourtytwo,
> 
> Could you provide us with full logs (try using adb logcat | grep "AppsFlyer_"). We would like to see if the native callbacks are being called.
> Also, did you make sure to add:
> @override
> public void onNewIntent(Intent intent) {
> super.onNewIntent(intent);
> setIntent(intent);
> }
> If the launch mode is [non-standard launch mode](https://developer.android.com/guide/topics/manifest/activity-element#lmode) (which is single by default in flutter).

i guess this is caused by this, sorry for the late reply, was on another project.
i couldn't find any info how to add this override in my project, i'm not proficient with native android. so, if you could help,it would be appreciated, language of main class is kotlin, launchMode="singleTop".

**Comment by GM-appsflyer** [comment]
Hey @misterfourtytwo ,
Just try adding the following code in the activity class where you're deep linking into:
```
override fun onNewIntent(intent : Intent){
    setIntent(intent)
}
```

> The "standard" and "singleTop" modes differ from each other in just one respect: Every time there's a new intent for a "standard" activity, a new instance of the class is created to respond to that intent. Each instance handles a single intent. Similarly, a new instance of a "singleTop" activity may also be created to handle a new intent. However, if the target task already has an existing instance of the activity at the top of its stack, that instance will receive the new intent (in an onNewIntent() call); a new instance is not created. In other circumstances — for example, if an existing instance of the "singleTop" activity is in the target task, but not at the top of the stack, or if it's at the top of a stack, but not in the target task — a new instance would be created and pushed on the stack.

Try the following articles on the android launch modes: https://developer.android.com/guide/topics/manifest/activity-element.html#lmode

Please let us know if this works

**Comment by oyzxchi** [comment]
I can only get information from onInstallConversionData( ), and the information is always the following:
![image](https://user-images.githubusercontent.com/78333154/114143847-3a9c9480-9947-11eb-833f-e17e6b84331c.png)

**Comment by oyzxchi** [comment]
The corresponding version in pubspec.yaml is appsflyer_sdk: ^6.2.4+1-flutterv1
flutter doctor:
![image](https://user-images.githubusercontent.com/78333154/114256326-25794180-99eb-11eb-810a-d0c84197d9d6.png)

main.dart:
![image](https://user-images.githubusercontent.com/78333154/114256383-78eb8f80-99eb-11eb-90db-1792a80ee8bc.png)

**Comment by wangxingxing123654** [comment]
any update ??
 still has problem when app is terminated and reopened the app

**Comment by GM-appsflyer** [comment]
Issue fixed in v6.2.6-nullsafety.1
